src/test/java/lambda/part2/exercise/ArrowNotationExercise.java

```diff
@@ -14,35 +14,36 @@
     @Test
     public void getAge() {
         // Person -> Integer
-        final Function<Person, Integer> getAge = null; // TODO
+        final Function<Person, Integer> getAge = Person::getAge;
 
         assertEquals(Integer.valueOf(33), getAge.apply(new Person("", "", 33)));
     }
 
     @Test
     public void compareAges() {
-        // TODO use BiPredicate
+        // use BiPredicate
         // compareAges: (Person, Person) -> boolean
 
-        throw new UnsupportedOperationException("Not implemented");
-        //assertEquals(true, compareAges.test(new Person("a", "b", 22), new Person("c", "d", 22)));
+        BiPredicate<Person, Person> compareAges = (p1, p2) -> p1.getAge() == p2.getAge();
+        assertEquals(true, compareAges.test(new Person("a", "b", 22), new Person("c", "d", 22)));
     }
 
-    // TODO
+    //
     // getFullName: Person -> String
 
-    // TODO
+    //
     // ageOfPersonWithTheLongestFullName: (Person -> String) -> (Person, Person) -> int
     //
 
     @Test
     public void getAgeOfPersonWithTheLongestFullName() {
         // Person -> String
-        final Function<Person, String> getFullName = null; // TODO
+        final Function<Person, String> getFullName = person -> person.getFirstName() + " " + person.getLastName();
 
         // (Person, Person) -> Integer
-        // TODO use ageOfPersonWithTheLongestFullName(getFullName)
-        final BiFunction<Person, Person, Integer> ageOfPersonWithTheLongestFullName = null;
+        // use ageOfPersonWithTheLongestFullName(getFullName)
+        final BiFunction<Person, Person, Integer> ageOfPersonWithTheLongestFullName =
+                (p1, p2) -> getFullName.apply(p1).compareTo(getFullName.apply(p2)) > 0 ? p1.getAge() : p2.getAge();

```

Comment:
Use the method implemented above this way: `ageOfPersonWithTheLongestFullName(getFullName)`.